### ISSUE TITLE
Random error 'socket hang up' (ECONNRESET) while performing most time consuming sfdx operations

### ISSUE THREAD

**Issue by busybox0** [forcedotcom/cli#1417]
### Summary
In recent weeks we see Increased number of random sfdx operation failures. The error may arise at any time while long operation is performed, i.e. `sfdx force:source:deploy, sfdx force:source:push, sfdx force:package:install` etc. 
The issue seems not related to Internet connection quality. It appears both in datacenters like AWS,  and on desktops, with stable Internet connection. On average, around  2 - 5% of operations fail with following error:
```
time sfdx force:source:deploy --targetusername username -p force-app/ -w 90

*** Deploying with SOAP API ***
Deploy ID: 0Af9E0000...
SOURCE PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/8222 Components
...
SOURCE PROGRESS | ████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 1629/8222 Components
SOURCE PROGRESS | ████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 1629/8222 Components
Error: socket hang up
    at connResetException (internal/errors.js:639:14)
    at TLSSocket.socketOnEnd (_http_client.js:499:23)
    at TLSSocket.emit (events.js:412:35)
    at endReadableNT (internal/streams/readable.js:1334:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  code: 'ECONNRESET'
}
SOURCE PROGRESS | ████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 1629/8222 Components
ERROR running force:source:deploy:  Metadata API request failed: socket hang up
real	6m51.253s
user	0m18.841s
sys	0m3.163s
```

**Comment by mshanemc** [comment]
You didn't complete the issue template section where you run `sfdx version --verbose`.  That's important for this issue.

**Comment by stoevie** [comment]
Hello, we had the same problem frequently last week with the sfdx force:source:push command.

Error: socket hang up
    at connResetException (internal/errors.js:609:14)
    at TLSSocket.socketOnEnd (_http_client.js:458:23)
    at TLSSocket.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  code: 'ECONNRESET'
}
(node:3269) UnhandledPromiseRejectionWarning: Error: socket hang up
    at connResetException (internal/errors.js:609:14)
    at TLSSocket.socketOnEnd (_http_client.js:458:23)
    at TLSSocket.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)

**SFDX version**

	linux-x64

 Node Version : 
	node-v12.22.10

 Plugin Version: 
	@oclif/plugin-autocomplete 0.3.0 (core)
	@oclif/plugin-commands 1.3.0 (core)
	@oclif/plugin-help 3.3.1 (core)
	@oclif/plugin-not-found 1.2.6 (core)
	@oclif/plugin-plugins 1.10.11 (core)
	@oclif/plugin-update 1.5.0 (core)
	@oclif/plugin-warn-if-update-available 1.7.3 (core)
	@oclif/plugin-which 1.0.4 (core)
	@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
	alias 1.2.1 (core)
	apex 0.8.0 (core)
	auth 1.8.1 (core)
	community 1.1.4 (core)
	config 1.3.19 (core)
	custom-metadata 1.0.12 (core)
	data 0.6.9 (core)
	generator 1.2.2 (core)
	info 1.2.1 (core)
	limits 1.3.0 (core)
	org 1.11.1 (core)
	salesforce-alm 53.10.2 (core)
	schema 1.1.0 (core)
	sfdx-cli 7.138.1 (core)
	source 1.8.11 (core)
	telemetry 1.4.0 (core)
	templates 53.5.0 (core)
	trust 1.1.0 (core)
	user 1.7.1 (core)

**Comment by MaciekLeks** [comment]
Are there any updates on this issue? We face it every day, sometimes more, sometimes less. It's become the most constant thing in our system. Now, the problem is even bigger.  We also use a Squid proxy with parent-child setup. Here's what we noticed and what might be causing it:

1. The server sends empty TLS messages, and some clients might read EOF as TCP RST.
2. A standard node client without good agent (e.g. global-agent) can't handle the RST.
